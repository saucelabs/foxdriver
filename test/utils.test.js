import { spy } from 'sinon'

import {
    transformEvaluateArgs, transformEvaluateScript, addEventListener, removeEventListeners
} from '../lib/utils'

test('transform script properly', () => {
    expect(transformEvaluateScript('foo', 'someargs')).toEqual(
        '(function () { foo }).apply(window, someargs)')
})

test('transform script function properly', () => {
    expect(transformEvaluateScript(() => { return 'foobar' }, 'someargs')).toMatchSnapshot()
})

test('transform arguments properly', () => {
    const args = transformEvaluateArgs([() => 'foobar', 1, '2', false, [1, 2, 3], {foo: 'bar'}])
    expect(args).toEqual('["() => \'foobar\'",1,"2",false,[1,2,3],{"foo":"bar"}]')
})

test('transform arguments properly if not passed as array', () => {
    const args = transformEvaluateArgs('foobar')
    expect(args).toEqual('["foobar"]')
})

test('creates enhanced listener objects', () => {
    const emitter = { on: spy() }
    const handler = spy()
    addEventListener(emitter, 'foobar', handler)
    expect(emitter.on.calledWith('foobar', handler)).toBe(true)
})

test('unregisteres listener events', () => {
    const emitter = { removeListener: spy(), on: () => {} }
    const handler = spy()
    const listeners = [addEventListener(emitter, 'foobar', handler)]

    removeEventListeners(listeners)
    expect(emitter.removeListener.calledWith('foobar', handler)).toBe(true)
    expect(listeners).toHaveLength(0)
})
