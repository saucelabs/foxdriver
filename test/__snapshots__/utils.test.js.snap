// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform script function properly 1`] = `
"(() => {
    return 'foobar';
  }).apply(window, someargs)"
`;
